crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -844,7 +844,7 @@ where
                 range: _,
             }) => {
                 if let Expr::Name(ast::ExprName { id, ctx, range: _ }) = func.as_ref() {
-                    if id == "locals" && ctx.is_load() {
+                    if *id == "locals" && ctx.is_load() {
```

Comment:
Not sure off the top of my head, so let's run through it. `x == y` is syntactic sugar for [`std::cmp::PartialEq::eq(&x, &y)`](https://doc.rust-lang.org/reference/expressions/operator-expr.html#comparison-operators). If `id` is a `&CompactString` and `y` is a `&'static str`, then `&x` is actually an `&&CompactString` and `y` is actually a `&&str`. So I believe this means we need to look for a `PartialEq` impl for `&&CompactString` (notice the double `&` there).

The only `PartialEq` impl for `CompactString` is `impl<T: AsRef<str>> PartialEq<T> for CompactString`, which on its own does not fit here because we're looking for `&&CompactString`.

At least, I think that's why. It's possible the `compact-str` crate could add more impls to alleviate this. But maybe not.

Auto-deref also comes into the picture here. See:

* https://stackoverflow.com/questions/72909938/why-operator-requires-dereference-while-eq-does-not
* https://github.com/rust-lang/rust/issues/44762